Source/CMakeLists.txt

```diff
@@ -16,18 +16,13 @@ if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
   check_and_add_flag(CPPLATEST /std:c++latest)
   check_and_add_flag(STANDARD_COMPLIANCE /permissive-)
 else()
-  # Enable C++17, but fall back to C++14 if it isn't available.
-  # CMAKE_CXX_STANDARD cannot be used here because we require C++14 or newer, not any standard.
+  # Enable C++17
+  # CMAKE_CXX_STANDARD cannot be used because it requires cmake 3.8+.
```

Comment:
Overall looks good to me, but CMake's minimum should really be raised, in order to set the minimum through it, getting rid of the need to manually manage different compiler flags. Ubuntu 18.04 provides up to CMake 3.10 anyways.

Reply:
Done :)